### PR TITLE
feat: add more customization for labels

### LINKED ISSUE #71: [zabbix-community/zabbix] Add support for custom labels
**Is your feature request related to a problem? Please describe.**

We're trying to setup this chart on an AKS cluster using the secrets-store CSI. Besides [needing a label on the SA](https://github.com/zabbix-community/helm-zabbix/issues/70) we also need a label on the pod to be able to use this driver.
It would be nice if we could add custom labels to stuff like pod, sa, services etc.

**Is your feature request related to a problem? Please describe.**

See above

**Describe the solution you'd like**

[Cert-manager](https://github.com/cert-manager/cert-manager/blob/master/deploy/charts/cert-manager/values.yaml) for example has `podLabels` and `serviceAccount.labels` which gets applied to the SA and Pods it creates. 
There is also a `commonLabels` option.

It would be nice to have something like that here as well, but perhaps (also) a way to specify this for specific resources (e.g. only `zabbix-server` pod)

**Describe alternatives you've considered**
-

**Additional context**
Related issue: #70

### PR DESCRIPTION
**Pull request by syphernl**
#### What this PR does / why we need it:
Adds:
- `global.commonLabels` to globally add labels to all resources
- `deploymentLabels` and `containerLabels` to add labels to Agent, Server, Web & WebService

#### Which issue this PR fixes
  - fixes #71

#### Checklist

- [X] [DCO](https://github.com/zabbix-community/helm-zabbix/blob/master/CONTRIBUTING.md) signed
- [ ] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

File: charts/zabbix/templates/deployment-zabbix-server.yaml [hunk 1]
```diff
@@ -31,6 +34,9 @@ spec:
         helm.sh/chart: {{ include "zabbix.chart" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}-zabbix-server
         app.kubernetes.io/managed-by: {{ .Release.Service }}-zabbix-server
+        {{- if .Values.zabbixServer.containerLabels }}
+        {{- toYaml .Values.zabbixServer.containerLabels | nindent 8 }}
```
Comment (aeciopires) [inline charts/zabbix/templates/deployment-zabbix-server.yaml:38]: Hi @syphernl!
 
I think the ZabbixAgent container labels can also be included here when it is not in daemonSet mode. Do you agree?

File: charts/zabbix/values.yaml [hunk 2]
```diff
@@ -2,6 +2,18 @@
...
+# Global configurations
+global:
+  # Reference to one or more secrets to be used when pulling images.
+  #
+  # For example:
+  #  imagePullSecrets:
+  #    - name: "image-pull-secret"
+  imagePullSecrets: []
+
+  # Labels to apply to all resources.
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:14]: Hi @syphernl!

Can you change this line?

`# -- Labels to apply to all resources.`

File: charts/zabbix/values.yaml [hunk 3]
```diff
@@ -343,8 +361,12 @@ zabbixAgent:
   extraVolumeMounts: []
   # -- annotations to add to the daemonSet
   daemonSetAnnotations: {}
+  # -- labels to add to the daemonSet
+  daemonSetLabels: {}
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:365]: Hi @syphernl!

Could you create the deploymentLabel to use in the Zabbix Server and Zabbix Proxy template when the ZabbixAgent is not in daemonSet mode?

```
# -- labels to add to the deploymentSet
deploymentLabels: {}
```

**Review by aeciopires — CHANGES_REQUESTED** [review]
Hi @syphernl!

Thank you very much for your contribution!

I really enjoyed seeing your PR.

I missed creating custom Labels for Zabbix Proxy and PostgreSQL. Could you implement this too?

Another thing... I had PR pending to be merged... I'm going to merge it now and publish a version. It will probably cause a conflict in the contents of a file... Can you resolve them if this happens?

You will also need to increase the chart version in the Chart.yaml file

**Comment by syphernl** [comment]
@aeciopires I made the requested changes.

**Review by aeciopires — CHANGES_REQUESTED** [review]
Hi @syphernl 

All that remains is to include the Zabbix Agent deployment and container labels in the agent container-specific part within the Zabbix Server deployment template and Zabbix Proxy StatefulSet files

**Comment by aeciopires** [comment]
Hi @syphernl 

I merged a PR that was pending, but I haven't published a version yet... I'll wait for your changes...

Could you synchronize your 2 PRs with the master branch to get my changes and correct any conflicts?

**Comment by syphernl** [comment]
I pulled in the changes of master into this branch. No conflicts to resolve.

I can pick up further changes next week, or feel free to add these yourself 😄

**Comment by aeciopires** [comment]
Wonder! It's great this way. I will do the merge and a few more small adjustments and publish the new version of the chart today. Here in Brazil it is early afternoon.

Thank you for your contribution

**Pull request merged**